lib/benchmark/bench.go

```diff
@@ -52,98 +63,91 @@ type BenchmarkResult struct {
 	Histogram *hdrhistogram.Histogram
 	// LastError contains last recorded error
 	LastError error
+	// Duration it takes for the whole benchmark to run
+	Duration time.Duration
 }
 
 // Benchmark connects to remote server and executes requests in parallel according
 // to benchmark spec. It returns benchmark result when completed.
 // This is a blocking function that can be cancelled via context argument.
-func (tc *TeleportClient) Benchmark(ctx context.Context, bench Benchmark) (*BenchmarkResult, error) {
+func (c *Config) Benchmark(ctx context.Context, tc *client.TeleportClient) (Result, error) {
 	tc.Stdout = ioutil.Discard
 	tc.Stderr = ioutil.Discard
 	tc.Stdin = &bytes.Buffer{}
+	workerCtx, cancelWorkers := context.WithCancel(ctx)
```

Comment:
rename `workerCtx` to `ctx` (reuse the same name)
this ensures that you use the cancel-able context everywhere below, and not accidentally the original one